modules/dns_common/dns_common.rb

```diff
@@ -15,12 +17,20 @@ def resolver
       Resolv::DNS.new(:nameserver => @server)
     end
 
-    def dns_find key
-      if match = key.match(/(\d{1,3})\.(\d{1,3})\.(\d{1,3})\.(\d{1,3})/)
-        resolver.getname(match[1..4].reverse.join(".")).to_s
-      else
-        resolver.getaddress(key).to_s
+    def dns_find key, type='auto'
+      if type == 'auto'
```

Comment:
I'm suggesting braking up https://github.com/ekohl/smart-proxy/blob/aaaa-records/modules/dns_common/dns_common.rb#L20 into two methods, something along these lines:

```
def calculate_key_type(key, type)
  key, type = case key
  when Resolv::IPv4::Regex
    [Resolv::IPv4.create(key).to_name.to_s, 'PTR']
  when Resolv::IPv6::Regex
    [Resolv::IPv6.create(key).to_name.to_s, 'PTR']
  else
    [key, 'A']
  end if key == 'auto'
	
  [key, type]
end

def dns_find(key, type='auto')
  resolver.getresource(*calculate_key_type(key, type)).to_s
end
```

Then you can test the "interesting" logic in calculate_key_type without any mocking. Probably worth adding one integration-type test to verify that dns_find calls calculate_key_type and resolver.getresource though.